### DISCUSSION TITLE
qm.Load does not work as expected if the foreign key column isn't in qm.Select

### DISCUSSION THREAD

**Discussion by vladvelici** [aarondl/sqlboiler#893] [Q&A]
Hi! I'm not sure if this is a bug or just something to be careful with, so I'm not opening an issue. I have roughly this schema:

    Location (id, address_id not null)
    Address (id)

There are many other columns, but irrelevant here. Foreign key set as expected, location.address_id references address.id.

This kind of query won't return any address:

    one, err := models.Locations(
        qm.Select(models.LocationColumns.ID),
	qm.Load(models.LocationRels.Address),
    ).OneG()

one.R.Address is simply nil. I'm guessing this is because eager loading does a second query using the one.AddressID for the subsequent WHERE IN statement?

The above query works just fine, address fetched as expected.

    one, err := models.Locations(
        qm.Select(models.LocationColumns.ID, models.LocationColumns.AddressID),
	qm.Load(models.LocationRels.Address),
    ).OneG()

Is this intended behaviour? Should there be a check to prevent such cases? Either auto-correct and fetch the foreign key column or return an error?

**Comment by aarondl** [comment]
This is just sort of more badness associated with the way eager loading works. I don't have a good answer for you. It's hard to introspect a query struct because it's all unexported madness. So it's hard to auto-detect or auto-correct. Expected behavior? It shouldn't be but that's where we're at with eager loading atm :(
